### DISCUSSION TITLE
Custom Error Page not rendered when trowing an error using `createError` inside Promise.

### DISCUSSION THREAD

**Discussion by marcoborsoi** [nuxt/nuxt#21703] [Questions]
Hi!

I want to display a custom 404 page when a GraphQL query does not return the requested data. I have created the error.vue in root following the docs here: https://nuxt.com/docs/getting-started/error-handling#rendering-an-error-page

I'm using __@nuxtjs/apollo__ (https://apollo.nuxtjs.org/) and I'm asking the same question there as well.

The issue I'm facing is that `createError` throws a `[nitro] [dev] [uncaughtException] H3Error` and the custom error page (error.vue) is not displayed.
In my particular case, the issue only happens when I use `createError` inside `onResult`.

Here's the reproduction using Nuxt Apollo: https://stackblitz.com/edit/nuxt-apollo-demo-cd35uc?file=app.vue

---

The same issue happens when trowing an error using `createError` inside  a `Promise`: `createError` throws a `[nitro] [dev] [uncaughtException] H3Error` and the custom error page (error.vue) is not displayed.

Here's the reproduction using Promises: https://stackblitz.com/edit/nuxt-starter-w4jdc1?file=app.vue

Thank you

**Comment by AasmundN** [comment]
I believe I have experienced something similar. Throwing an error inside the `onResponseError` handler in useFetch does not render the error page.

**Reply by marcoborsoi** [reply]
@AasmundN I've created the issue https://github.com/nuxt/nuxt/issues/21848 for this, where @danielroe explained that in this case we should use the `showError` composable.

I've test it on my side and it works on my @nuxt/apollo example (https://stackblitz.com/edit/nuxt-apollo-demo-cd35uc?file=app.vue,error.vue,package.json), but not on the one using a Promise (https://stackblitz.com/edit/nuxt-starter-w4jdc1?file=app.vue,error.vue,package.json). I may be missing something.

**Comment by xjccc** [comment]
its the promise error?
throw error like this, its not async
```
const {data} = await mock(false);
if(!true) {
  throw createError({
    statusMessage: 'Error during promise',
    statusCode: 404,
    fatal: true,
  });
}
```
